### PR TITLE
feat: add "subjectPriority(p.eft) || deny" feature

### PR DESCRIPTION
**Pull request by tangyang9464**
Signed-off-by: tangyang9464 <tangyang9464@163.com>

fix: #252

### CODE REVIEW

File: docs/PriorityModel.md [hunk 1]
```diff
@@ -61,4 +61,79 @@ request:
...
 bob, data2, read --> false
 bob, data2, write --> true // for bob has role of `data2_allow_group` which has right to write data2, and there's no deny policy with higher priority 
-```
\ No newline at end of file
+```
+## Load Policy with Priority Based on Role and User Hierarchy
+
+> Also see: [casbin#831](https://github.com/casbin/casbin/issues/831)
+
+The inherited structure of roles and users can only be multiple trees, not graphs.
```
Comment (hsluoyz) [inline docs/PriorityModel.md:69]: Put these lines into a paragraph. Don't use multiple single sentences.

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
@ErikQQY plz review carefully, especially for grammar errors, link errors, etc.

File: docs/SupportedModels.md [hunk 2]
```diff
@@ -31,4 +31,6 @@ RESTful | [keymatch_model.conf](https://github.com/casbin/casbin/blob/master/exa
...
 Allow-and-deny | [rbac_with_deny_model.conf](https://github.com/casbin/casbin/blob/master/examples/rbac_with_deny_model.conf)  | [rbac_with_deny_policy.csv](https://github.com/casbin/casbin/blob/master/examples/rbac_with_deny_policy.csv)
 Priority | [priority_model.conf](https://github.com/casbin/casbin/blob/master/examples/priority_model.conf)  | [priority_policy.csv](https://github.com/casbin/casbin/blob/master/examples/priority_policy.csv)
-Explicit Priority | [priority_model_explicit](https://github.com/casbin/casbin/blob/master/examples/priority_model_explicit.conf) | [priority_policy_explicit.csv](https://github.com/casbin/casbin/blob/master/examples/priority_policy_explicit.csv)
\ No newline at end of file
+Explicit Priority | [priority_model_explicit](https://github.com/casbin/casbin/blob/master/examples/priority_model_explicit.conf) | [priority_policy_explicit.csv](https://github.com/casbin/casbin/blob/master/examples/priority_policy_explicit.csv)
+Subject-Priority | [subject_priority_model.conf](https://github.com/casbin/casbin/blob/master/examples/subject_priority_model.conf) | [subject_priority_model.csv](subject_priority_model.csv) 
+
```
Comment (ErikQQY) [inline docs/SupportedModels.md:36]: The subject_priority_model.csv link doesn't work

File: docs/SyntaxForModels.md [hunk 3]
```diff
@@ -91,6 +91,7 @@ some(where (p.eft == allow)) | allow-override | [ACL, RBAC, etc.](/docs/en/suppo
 !some(where (p.eft == deny)) | deny-override | [Deny-override](/docs/en/supported-models#examples)
 some(where (p.eft == allow)) && !some(where (p.eft == deny)) | allow-and-deny | [Allow-and-deny](/docs/en/supported-models#examples)
 priority(p.eft) &#124;&#124; deny | priority | [Priority](/docs/en/supported-models#examples)
+subjectPriority(p.eft) | priority base on role | [Subject-Priority](https://deploy-preview-254--xenodochial-leakey-a5c0e8.netlify.app/docs/en/supported-models#examples)
```
Comment (ErikQQY) [inline docs/SyntaxForModels.md:95]: To link to other docs, please refer to other examples, instead of using netlify links.

**Review by ErikQQY — CHANGES_REQUESTED** [review]

**Comment by tangyang9464** [comment]
@ErikQQY fixed

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**